wpilibcExamples/src/main/cpp/examples/StateSpaceDifferentialDriveSimulation/cpp/subsystems/DriveSubsystem.cpp

```diff
@@ -0,0 +1,120 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "subsystems/DriveSubsystem.h"
+
+#include <frc/RobotController.h>
+#include <frc/SPI.h>
+#include <frc/geometry/Rotation2d.h>
+#include <frc/kinematics/DifferentialDriveWheelSpeeds.h>
+#include <frc/simulation/SimDeviceSim.h>
+
+using namespace DriveConstants;
+
+DriveSubsystem::DriveSubsystem()
+    : m_left1{kLeftMotor1Port},
+      m_left2{kLeftMotor2Port},
+      m_right1{kRightMotor1Port},
+      m_right2{kRightMotor2Port},
+      m_leftEncoder{kLeftEncoderPorts[0], kLeftEncoderPorts[1]},
+      m_rightEncoder{kRightEncoderPorts[0], kRightEncoderPorts[1]},
+      m_gyro(0),
+      m_odometry{m_gyro.GetRotation2d()},
+      m_drivetrainSimulator{kDrivetrainPlant, kTrackwidth, kDrivetrainGearbox,
+                            kDrivetrainGearing,
+                            units::inch_t(kWheelDiameterInches / 2)},
+      m_leftEncoderSim(m_leftEncoder),
+      m_rightEncoderSim(m_rightEncoder),
+      m_gyroAngleSim(m_gyro) {
```

Comment:
It would be a lot easier to read if these were initialized in the header instead.